src/imputation/MoR.py

```diff
@@ -19,17 +21,25 @@ def run_mor(df, backdata, impute_vars):
         df (pd.DataFrame): Processed full responses DataFrame
         backdata (pd.DataFrame): One period of backdata.
         impute_vars ([string]): List of variables to impute.
+        lf_target_vars ([string]): List of long form target vars.
 
     Returns:
-        pd.DataFrame: df with MoR applied
+        pd.DataFrame: df with MoR applied.
+        pd.DataFrame: QA DataFrame showing how imputation links are calculated.
     """
     to_impute_df, remainder_df, backdata = mor_preprocessing(df, backdata)
 
     # Carry forwards method
     carried_forwards_df = carry_forwards(to_impute_df, backdata, impute_vars)
 
+    gr_df = calculate_growth_rates(remainder_df, backdata, lf_target_vars)
+    links_df = calculate_links(gr_df, lf_target_vars, config)
+
+    carried_forwards_df = apply_links(
+        carried_forwards_df, links_df, lf_target_vars, config
+    )
     # TODO Remove the `XXX_prev` columns (left in for QA)
-    return pd.concat([remainder_df, carried_forwards_df]).reset_index(drop=True)
+    return pd.concat([remainder_df, carried_forwards_df]).reset_index(), links_df
```

Comment:
I think you need to put that drop=True back in to the reset index, otherwise it creates an extra column called index (and that's even though we have index=False in the write_csv function)

Reply:
Not sure why I got rid of it in the first place! Will do.